pandas/tests/groupby/test_groupby.py

```diff
@@ -1825,6 +1825,18 @@ def test_groupby_empty_list_raises():
         df.groupby([[]])
 
 
+def test_groupby_with_empty_category():
+    # GH-9614
+    df = pd.DataFrame({"id": [None] * 3, "spam": [None] * 3})
+    df["spam"] = df["spam"].astype("category")
+    df["spam"] == "spam"  # works as expected
```

Comment:
can you assert this result here

Reply:
I've refactored to what I think is more clear & explicit. 
Please approve fine or let me know more asserts still needed, thanks!